dra-evolution/pkg/api/capacity_types.go

```diff
@@ -3,168 +3,110 @@ package api
+	// The name of each attribute must be unique.
 	//
+	// +listType=atomic
 	// +optional
-	SharedResourcesConsumed map[string]resource.Quantity `json:"sharedResourcesConsumed,omitempty"`
+	Attributes []DeviceAttribute `json:"attributes,omitempty" protobuf:"bytes,3,opt,name=attributes"`
 
-	// ClaimResourcesProvided allows the definition of per-device resources
-	// that can be allocated in a manner similar to standard Kubernetes
-	// resources.
-	//
-	// +optional
-	ClaimResourcesProvided []ResourceCapacity `json:"claimResourcesProvided,omitempty"`
+	// TODO for 1.31: define how to support partitionable devices
 }
 
-type ResourceCapacity struct {
-	// Name is the resource name/type.
-	// +required
-	Name string `json:"name"`
-
-	// Capacity is the total capacity of the named resource.
-	// +required
-	Capacity resource.Quantity `json:"capacity"`
-
-	// BlockSize is the increments in which capacity is consumed. For
-	// example, if you can only allocate memory in 4k pages, then the
-	// block size should be "4Ki". Default is 1.
+// DeviceAttribute is a combination of an attribute name and its value.
+type DeviceAttribute struct {
+	// Name is a unique identifier across all possible attributes of devices.
+	// It must be a DNS subdomain, with one additional restriction: the
+	// first part must not contain a hyphen and not start with a digit.
 	//
-	// If the resource is consumable in a fractional way, then the
-	// default of 1 should not be used; instead this should be a fractional
-	// amount corresponding the increment size. We may also need a minimum
-	// value, if the minimum is larger than the block size (as is the case
-	// for standard Kubernetes CPU resources).
+	// If this is a DNS label (no dot), then the driver name gets added
+	// when looking up attributes. This avoids name collisions with attributes
+	// used by other drivers.
 	//
-	// +optional
-	BlockSize *resource.Quantity `json:"blockSize,omitempty"`
-}
+	// If this is a full DNS subdomain, then the meaning of the attribute is driver-independent.
+	// For example, Kubernetes will use `*.k8s.io` names when defining attributes that
+	// drivers from different vendors are supposed to use.
+	Name string `json:"name" protobuf:"bytes,1,name=name"`
 
-// Attribute capture the name, value, and type of an device attribute.
-type Attribute struct {
-	Name string `json:"name"`
-
-	// One of the following:
-	StringValue   *string            `json:"stringValue,omitempty"`
-	IntValue      *int               `json:"intValue,omitempty"`
-	QuantityValue *resource.Quantity `json:"quantityValue,omitempty"`
-	SemVerValue   *SemVer            `json:"semVerValue,omitempty"`
+	DeviceAttributeValue `json:",inline" protobuf:"bytes,2,opt,name=attributeValue"`
 }
 
-func (a Attribute) Equal(b Attribute) bool {
-	if a.Name != b.Name {
-		return false
-	}
-
-	return a.EqualValue(b)
+// The Go field names below have a Value suffix to avoid a conflict between the
```

Comment:
We would not really want this comment in the API docs

Reply:
Nor do I :smile:

It is a separate comment that does not get applied to DeviceAttributeValue when generating docs.